Exercise_1.java

```diff
@@ -1,33 +1,84 @@
+// Time Complexity :
+//      isEmpty() - O(1)
+//      push() - O(1)
+//      pop() - O(1)
+//      peek() - O(1)
+//      
+// Space Complexity :
+//      overall - O(1) - constant max size of the stack/array
+//      isEmpty() - N/A
+//      push() - N/A
+//      pop() - N/A
+//      peek() - N/A
+// Did this code successfully run on Leetcode : N/A
+// Any problem you faced while coding this : No
+
 class Stack { 
     static final int MAX = 1000; 
     int top; 
     int a[] = new int[MAX]; // Maximum size of Stack 
   
     boolean isEmpty() 
     { 
-        //Write your code here 
+        //Write your code here
+        //check if top is -1
+        if(top == -1)
+            return true;
+        return false;
     } 
 
     Stack() 
     { 
-        //Initialize your constructor 
+        //Initialize your constructor
+        //assign -1 to top
+        top = -1; 
     } 
   
     boolean push(int x) 
     { 
         //Check for stack Overflow
         //Write your code here
+        //check if top is equal to MAX
+        if(top == MAX)
+        {
+            System.out.println("Stack Overflow");
+            return false;
+        }

```

Comment:
This condition will given an exception at index already at 999 you will try to add an element to top = 1000 
The if statement should be for top == MAX-1

Reply:
Yes. Thanks for the review. Have committed the changes to the repo